### DISCUSSION TITLE
session data availability on context.startSession(...)

### DISCUSSION THREAD

**Discussion by perlohmann** [keystonejs/keystone#7577] [Questions] [answered]
Are there a good way to load session data into the system on the same request?
I have a scenario where I have implemented automated logging using the tables `afterOperation` hook. The tables are not public accessible, but requires you to be logged in.

I have a scenario where I want to redeem a token and assign a user to a project in the same call.
I can do that just fine with elevated context `context.sudo()`, no worries, but now the automated logs are not updated correctly since access to the fields requires the session to have a `session.data.id` which will not be there until the next request.
I am using the startSession method on the context to signin the user:

```typescript
await elevatedContext.startSession({
  listKey: "User",
  itemId: user.id,
});
```

So the issue I am experiencing, is because when using the startSession the data populated in the normal auth process is not populated/available before the next call.
The messages that I am getting in my log are empty because you cannot access the table fields, because you are not really authenticated until the next request on the server.

// Per

**Comment by raveling** [comment, marked as answer]
@perlohmann you can do `context.session = ...`

Make sure to do this on the regular context, not the elevated context.

**Reply by perlohmann** [reply]
Thx @raveling. This helped a lot. 

One note though: when using the elevated context you should set the elevated context session and not the regular context session in order to have the data carry to the hooks (sort of obvious, but a little contradictory to your answer). 
My solution was to do the following:
1) add session with proper item id to the elevated session.
2) perform elevated queries.
3) use startSession on the regular context (for future requests).
